### ISSUE TITLE
Consider adding UTCTime and UUID instance for DynamoEncodable and DynamoScalar

### ISSUE THREAD

**Issue by paparga** [ondrap/dynamodb-generic#1]
Hi!
Great library, we are currently using in stagging and we want to eliminate orphan instance warning.
I could make a PR.

Thanks!

**Comment by ondrap** [comment]
If you can make a PR, I'll add it asap to the official version :)

**Comment by ondrap** [comment]
On the other hand, I wouldn't like to add additional dependencies. It seems that the library already depends on `time`, so I think that's a good candidate. However it doesn't depend on `uuid` so I'm not sure if that's a good idea.

**Comment by paparga** [comment]
ok, I will make the PR only for UTCTime this weekend

**Comment by paparga** [comment]
Sorry for delay, I looked to `aeson` [package](https://hackage.haskell.org/package/aeson) and it also depends on `uuid-types`. Would you reconsider adding UUID instance?

**Comment by ondrap** [comment]
Ok, I think that's OK, as UUIDs are generally supposed to be used a lot in dynamodb. Could you make it a separate PR?

I actually have much bigger problem with the UTCTime - it occurred to me now when I looked into your code. There are multiple ways to encode time. You have chosen to to encode it into posixseconds, however a pretty established scheme is to use the ISO time format as well - e.g. I use these instances in my code:

```
instance DynamoEncodable UTCTime where
  dEncode time =
      case toJSON time of
          AE.String str -> dEncode str
          _ -> error "Aeson toJSON didn't return a string for UTCTime"
  dDecode attr = do
      txt <- dDecode attr
      scalarDecode (ScS txt)

instance DynamoScalar 'D.S UTCTime where
  scalarEncode time =
      case toJSON time of
          AE.String str -> ScS str
          _ -> error "Aeson toJSON didn't return a string for UTCTime"
  scalarDecode (ScS txt) = AE.parseMaybe parseJSON (AE.String txt)
```
and it seems to me that is valid as well..

**Comment by paparga** [comment]
OK, I will make a separate PR and I will look a good implementation for it.

Regarding UTCTime I just used the implementation that it is in examples folder, if there is a better one lets do it!

BTW this is my first contribution to Haskell, so thank you for your time!

**Comment by paparga** [comment]
**Testing**

I noticed that Travis failed, how can I fix that?

And also, do you want to add some test about this instances?

**Comment by ondrap** [comment]
I have added the UUID instance and also fixed the problem with compiling with 8.2 (so you should have one orphan less). However, I am unsure about the `UTCTime` instance. It seems to me that the one above converting the result the same as JSON would would actually be better.

**Comment by paparga** [comment]
Thanks!
Yeah, we are also changing to converting the result the same as JSON.
